src/core/properties/cameraproperty.cpp

```diff
@@ -53,19 +54,45 @@ CameraProperty::CameraProperty(const std::string& identifier, const std::string&
                                vec3 center, vec3 lookUp, InvalidationLevel invalidationLevel,
                                PropertySemantics semantics)
     : CompositeProperty{identifier, displayName, invalidationLevel, semantics}
-    , cameraType_("cameraType", "Camera Type",
-                  InviwoApplication::getPtr()->getCameraFactory()->getKeys(), 0)
+    , factory_{InviwoApplication::getPtr()->getCameraFactory()}
+    , cameraType_("cameraType", "Camera Type", factory_->getKeys(),
+                  [&]() {
+                      auto keys = factory_->getKeys();
+                      auto it = std::find(keys.begin(), keys.end(), "PerspectiveCamera");
+                      return it != keys.end() ? std::distance(keys.begin(), it) : 0;
+                  }())
+    , camera_{factory_->create(cameraType_)}
+    , defaultCamera_{}
     , cameraActions_("actions", "Actions", buttons())
-    , lookFrom_("lookFrom", "Look from", eye, -vec3(100.0f), vec3(100.0f), vec3(0.1f),
-                InvalidationLevel::InvalidOutput, PropertySemantics{"SphericalSpinBox"})
-    , lookTo_("lookTo", "Look to", center, -vec3(100.0f), vec3(100.0f), vec3(0.1f),
-              InvalidationLevel::InvalidOutput, PropertySemantics::SpinBox)
-    , lookUp_("lookUp", "Look up", lookUp, -vec3(1.f), vec3(1.f), vec3(0.1f),
-              InvalidationLevel::InvalidOutput, PropertySemantics::SpinBox)
-    , aspectRatio_("aspectRatio", "Aspect Ratio", 1.0f, 0.0f, std::numeric_limits<float>::max(),
-                   0.01f, InvalidationLevel::InvalidOutput, PropertySemantics::Text)
-    , nearPlane_("near", "Near Plane", 0.1f, 0.001f, 10.f, 0.001f)
-    , farPlane_("far", "Far Plane", 100.0f, 1.0f, 1000.0f, 1.0f)
+    , lookFrom_(
+          "lookFrom", "Look from", [this]() { return camera_->getLookFrom(); },
+          [this](const vec3& val) { camera_->setLookFrom(val); },
+          {-vec3(100.0f), ConstraintBehavior::Ignore}, {vec3(100.0f), ConstraintBehavior::Ignore},
+          vec3(0.1f), InvalidationLevel::InvalidOutput, PropertySemantics{"SphericalSpinBox"})
```

Comment:
I hate the spherical spinbox :)
The rotations in that space does not make much sense for camera interaction

Reply:
xyz is not much better...

Reply:
maybe it would be better to use xyz and then add some more buttons (?) for various view changes? (zoom+-, various rotations)

Reply:
Definitely! Rotate left/right/up/down, translate left/right (perhaps forward/backward) and zoom in/out

Reply:
careful here! Zoom means you change the fov but not the position. You just ensure that an object at a certain distance, i.e. the look at, stays about the same size on the screen.
Moving the camera closer/further away is a dolly.

Reply:
on the other hand we really don't want to zoom using FOV since that will also be quite confusing in many cases...